jax/interpreters/partial_eval.py

```diff
@@ -150,12 +154,19 @@ def process_map(self, map_primitive, f, tracers, params):
     out_pvs = [None if pv is None else _unmapped_aval(params['axis_size'], pv)
                for pv in out_pvs_reduced]
     const_tracers = map(self.new_instantiated_const, consts)
+    env_tracers = map(self.full_raise, env)
     lifted_jaxpr = closure_convert_jaxpr(jaxpr)
-    bound_subjaxpr = (lifted_jaxpr, (), map(self.full_raise, env))
     out_tracers = [JaxprTracer(self, PartialVal((out_pv, out_pv_const)), None)
                    for out_pv, out_pv_const in zip(out_pvs, out_pv_consts)]
-    eqn = new_eqn_recipe(tuple(it.chain(const_tracers, tracers)),
-                         out_tracers, map_primitive, (bound_subjaxpr,), params)
+    # The `jaxpr` already contains the env_vars at start of invars
+    mapped_invars = tuple([True] * len(const_tracers) +
+                          [False] * len(env_tracers) +
+                          [True] * len(tracers))
```

Comment:
Instead of always `[True] * len(tracers)`, should this depend on `params['mapped_invars']`? In other words, should we use `params['mapped_invars']` here somewhere in this function? I think we're currently just ignoring its value, but that might mean we're turning some arguments (corresponding to the `tracers` here) from unmapped to mapped.

Reply:
This is a good question. `mapped_invars` here marks what before were the freevars, which were derived directly from `env_tracers` and not other information, and before, freevars were exactly the unmapped vars. This is why I think that this code is equivalent to the code before the change.